plugins/MauticFullContactBundle/Controller/PublicController.php

```diff
@@ -0,0 +1,313 @@
+                            $result['contactInfo']['websites']
+                        )) ? $result['contactInfo']['websites'][0]['url'] : '',
+                    'skype' => (array_key_exists('chats', $result['contactInfo']) && array_key_exists(
+                            'skype',
+                            $result['contactInfo']['chats']
+                        )) ? $result['contactInfo']['chats']['skype']['handle'] : '',
+                ];
+            }
+            $data = array_merge(
+                $data,
+                [
+                    'company'  => (null !== $org) ? $org['name'] : '',
+                    'position' => (null !== $org) ? $org['title'] : '',
+                    'city'     => (null !== $loc && array_key_exists('city', $loc) && array_key_exists(
+                            'name',
+                            $loc['city']
+                        )) ? $loc['city']['name'] : '',
+                    'state' => (null !== $loc && array_key_exists('state', $loc) && array_key_exists(
+                            'name',
+                            $loc['state']
+                        )) ? $loc['state']['name'] : '',
+                    'country' => (null !== $loc && array_key_exists('country', $loc) && array_key_exists(
+                            'name',
+                            $loc['country']
+                        )) ? $loc['country']['name'] : '',
+                ]
+            );
+
+            $data = array_merge($data, $social);
+
+            /** @var \Mautic\LeadBundle\Model\LeadModel $model */
+            $model = $this->getModel('lead');
+            /** @var Lead $lead */
+            $lead = $model->getEntity($id);
+            $model->setFieldValues($lead, $data);
+            $model->saveEntity($lead);
+
+            if ($notify && (!isset($lead->imported) || !$lead->imported)) {
+                /** @var UserModel $userModel */
+                $userModel = $this->getModel('user');
+                $user      = $userModel->getEntity($uid);
+                if ($user) {
+                    $this->addNewNotification(
+                        sprintf('The contact information for %s has been retrieved', $lead->getEmail()),
+                        'FullContact Plugin',
+                        'fa-search',
+                        $user
+                    );
+                }
+            }
+        } catch (\Exception $ex) {
+            try {
+                if ($notify && isset($lead, $uid) && (!isset($lead->imported) || !$lead->imported)) {
+                    /** @var UserModel $userModel */
+                    $userModel = $this->getModel('user');
+                    $user      = $userModel->getEntity($uid);
+                    if ($user) {
+                        $this->addNewNotification(
+                            sprintf(
+                                'Unable to save the contact information for %s: %s',
```

Comment:
This should be handled using a `$this->get('translator')->trans('some key', ['%email%' => $lead->getEmail(), '%message%' => $ex->getMessage()`